tests/Intl/Transliterator/TransliteratorTest.php

```diff
@@ -0,0 +1,529 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Tests\Intl\Transliterator;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Polyfill\Intl\Transliterator\Transliterator as p;
+
+/**
+ * @covers \Symfony\Polyfill\Intl\Transliterator\Transliterator::<!public>
+ */
+class TransliteratorTest extends TestCase
+{
+    private $old_mb_internal_encoding;
+
+    public function setUp()
+    {
+        if (function_exists('mb_internal_encoding')) {
+            $this->old_mb_internal_encoding = mb_internal_encoding();
+            mb_internal_encoding('UTF-8');
```

Comment:
this looks wrong to me. We should not write a polyfill assuming that the global state of mbstring is set to UTF-8, especially if as it is not the default on old PHP versions. We should instead pass an encoding explicitly in our polyfill code.

We cannot expect users of the polyfill to do that change (especially given it is useless for the non-polyfill implementation which is not based on mbstring).

Reply:
I added the encoding parameter into the polyfill code, so that it now also work for php 5.3.